homeassistant/components/binary_sensor/fibaro.py

```diff
@@ -0,0 +1,71 @@
+"""
+Support for Fibaro binary sensors.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.fibaro/
+"""
+import logging
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDevice, ENTITY_ID_FORMAT)
+from homeassistant.components.fibaro import (
+    FIBARO_CONTROLLER, FIBARO_DEVICES, FibaroDevice)
+
+DEPENDENCIES = ['fibaro']
+
+_LOGGER = logging.getLogger(__name__)
+
+SENSOR_TYPES = {
+    'com.fibaro.doorSensor':   ['Door', 'mdi:window-open', 'door'],
+    'com.fibaro.windowSensor': ['Window', 'mdi:window-open', 'window'],
+    'com.fibaro.smokeSensor': ['Smoke', 'mdi:smoking', 'smoke'],
+    'com.fibaro.FGMS001': ['Motion', 'mdi:run', 'motion'],
+    'com.fibaro.heatDetector': ['Heat', 'mdi:fire', 'heat'],
+}
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Perform the setup for Fibaro controller devices."""
+    add_entities(
+        [FibaroBinarySensor(device, hass.data[FIBARO_CONTROLLER])
+         for device in hass.data[FIBARO_DEVICES]['binary_sensor']], True)
+
+
+class FibaroBinarySensor(FibaroDevice, BinarySensorDevice):
+    """Representation of a Fibaro Binary Sensor."""
+
+    def __init__(self, fibaro_device, controller):
+        """Initialize the binary_sensor."""
+        self._state = None
+        FibaroDevice.__init__(self, fibaro_device, controller)
```

Comment:
Since it has two parent classes, I'm not sure your proposal is okay, since the two parents have different parameters for their init. (BinarySensor class has none, FibaroDevice has these parameter needs).

Reply:
Ok, I changed it. I'm coming from C/C++ and it gives me a bad feeling overall, as this feels like less defensive code, but I also see why it makes sense.